index.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const request = require('request-promise-native');
+const fs = require('fs');
+const Yaml = require('js-yaml');
+
+/**
+ * Loads the yaml configuration from a file
+ * @method loadYaml
+ * @param  {String}     path        File path
+ * @return {Promise}    Promise that resolves to the template as a config object
+ */
+function loadYaml(path) {
+    return new Promise(resolve =>
+        resolve(Yaml.safeLoad(fs.readFileSync(path, 'utf8'))));
+}
+
+/**
+ * Validates the template yaml
+ * @method validateTemplate
+ * @param   {Object}        config          Template config
+ * @return {Promise}         Resolves when template is valid/rejects when error or invalid
+ */
+function validateTemplate(config) {
+    const hostname = process.env.SD_API_URL || 'https://api.screwdriver.cd';
+
+    return request({
+        method: 'POST',
+        url: `${hostname}/v4/validator/template`,
+        auth: {
+            bearer: 'does not matter'
+        },
+        json: true,
+        body: {
+            yaml: JSON.stringify(config)
+        }
+    }).then((response) => {
+        if (response.errors.length > 0) {
+            throw new Error('Template has invalid format');
```

Comment:
Should we also return the response body here so the user can see what errors there are?